crates/net/downloaders/src/bitfinity_evm_client.rs

```diff
@@ -289,6 +339,55 @@ impl BitfinityEvmClient {
         }
         self
     }
+
+    /// Creates a new JSON-RPC client with retry functionality
+    ///
+    /// Tries the primary URL first, falls back to backup URL if provided
+    pub async fn client(config: RpcClientConfig) -> Result<EthJsonRpcClient<ReqwestClient>> {
```

Comment:
Should we use the retry strategy only when testing the client or on every remote call?

Reply:
ok